### ISSUE TITLE
"file:" prefix in jetty.base variable

### ISSUE THREAD

**Issue by serba** [eclipse/jetty.project#2556]
We recently upgraded Jetty in our project from 9.3.8 to 9.4.8 and noticed that our `extraClasspath` trick to re-use the same jars across different web applications stopped working due to what seems like a behavior change in `jetty.base` format. It looks like previously it was just a system path, whereas now it is prefixed with `file:`.

We were using the following trick:
```
  <Set name="extraClasspath">
    <Call class="java.lang.String" name="join">
      <Arg>,</Arg>
      <Arg>
        <Call class="java.nio.file.Files" name="readAllLines">
          <Arg>
            <Call class="java.nio.file.Paths" name="get">
              <Arg><Property name="jetty.base"/>/webapps/myapp-extra-classpath.txt</Arg>
              <Arg><Array type="java.lang.String"/></Arg>
            </Call>
          </Arg>
        </Call>
      </Arg>
    </Call>
  </Set>
```

After upgrade to Jetty 9.4.8 - this stopped working with exception that file `file:/fs/path//webapps/myapp-extra-classpath.txt` does not exist. 

It looks like the problem is in new `file:` prefix that wasn't there before.

I managed to work around this problem by stripping `file:` prefix with another call from XML configuration:

```
                <Call class="org.eclipse.jetty.util.StringUtil" name="replace">
                  <Arg><Property name="jetty.base"/>/webapps/myapp-extra-classpath.txt</Arg>
                  <Arg type="String">file:</Arg>
                  <Arg type="String"></Arg>
                </Call>
```

I noticed that there were some changes around `jetty.base` and that new variable “jetty.base.uri” was introduced in #1038 or #1039. Considering the fact that there’s now `jetty.base.uri` - shouldn’t `jetty.base` be just file path, without `file:` in front of it?

**Comment by joakime** [comment]
This cannot be replicated.

Not sure where you are getting that (System?) Property value/syntax from.
How do you start Jetty?

**Comment by serba** [comment]
I just 

* downloaded and expanded `jetty-distribution-9.4.8.v20171121`
* created `demo-base/test-classpath.txt` file
```
sh> cat demo-base/test-classpath.txt
demo-base/lib/ext/test-mock-resources-9.4.8.v20171121.jar
```
* Added the following configuration to `test.xml` file
```
diff -U3 demo-base/webapps/test.xml~ demo-base/webapps/test.xml
--- demo-base/webapps/test.xml~	2018-05-21 15:43:13.000000000 -0700
+++ demo-base/webapps/test.xml	2018-05-21 15:43:16.000000000 -0700
@@ -21,6 +21,22 @@
   <Set name="contextPath">/test</Set>
   <Set name="war"><Property name="jetty.webapps" default="."/>/test.war</Set>

+  <Set name="extraClasspath">
+    <Call class="java.lang.String" name="join">
+      <Arg>,</Arg>
+      <Arg>
+        <Call class="java.nio.file.Files" name="readAllLines">
+          <Arg>
+            <Call class="java.nio.file.Paths" name="get">
+              <Arg><Property name="jetty.base"/>/test-classpath.txt</Arg>
+              <Arg><Array type="java.lang.String"/></Arg>
+            </Call>
+          </Arg>
+        </Call>
+      </Arg>
+    </Call>
+  </Set>
+
```
* Started Jetty with `java -jar start.jar jetty.base=demo-base` (this is documented [here](https://www.eclipse.org/jetty/documentation/9.4.x/startup-base-and-home.html)) and got
```
2018-05-21 15:45:36.448:WARN:oejd.DeploymentManager:main: Unable to reach node goal: started
java.lang.reflect.InvocationTargetException
...
	at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:798)
...
Caused by:
java.nio.file.NoSuchFileException: file:/Users/alexey/Desktop/jetty-distribution-9.4.8.v20171121/demo-base/test-classpath.txt
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
...
	at org.eclipse.jetty.util.TypeUtil.call(TypeUtil.java:529)
	at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.call(XmlConfiguration.java:798)
```

If I change my `text.xml` snippet to strip `file:` prefix with 
```
  <Set name="extraClasspath">
    <Call class="java.lang.String" name="join">
      <Arg>,</Arg>
      <Arg>
        <Call class="java.nio.file.Files" name="readAllLines">
          <Arg>
            <Call class="java.nio.file.Paths" name="get">
              <Arg>
                <!-- jetty.base started having "file:" prefix after Jetty 9.4 upgrade -->
                <Call class="org.eclipse.jetty.util.StringUtil" name="replace">
                  <Arg><Property name="jetty.base"/>/test-classpath.txt</Arg>
                  <Arg type="String">file:</Arg>
                  <Arg type="String"></Arg>
                </Call>
              </Arg>
              <Arg><Array type="java.lang.String"/></Arg>
            </Call>
          </Arg>
        </Call>
      </Arg>
    </Call>
  </Set>
```
then it starts up fine:

```
sh> java -jar start.jar jetty.base=demo-base
...
2018-05-21 15:52:22.639:INFO:oejs.Server:main: Started @2117ms
```

**Comment by joakime** [comment]
Ah, I think I found it.

The problem has nothing to do with the `jetty.base` and `jetty.base.uri` efforts on Issues #1038 and #1039.
Instead, this is limited to just impacting the use of XmlConfiguration, from the changes in Issue #1715 and commit 76981a0a15633ad22a7d41f2ea8b8a1a5f79245d

The `Resource.toString()` is the one returning the `file:` scheme.

**Comment by joakime** [comment]
This is fixed, and is available on Jetty 9.4.11.v20180605
